src/WSEvents.cpp

```diff
@@ -911,6 +915,106 @@ void WSEvents::OnSceneItemVisibilityChanged(void* param, calldata_t* data) {
+	WSEvents* instance = static_cast<WSEvents*>(param);
+
+	obs_scene_t* scene = nullptr;
+	calldata_get_ptr(data, "scene", &scene);
+
+	obs_sceneitem_t* sceneItem = nullptr;
+	calldata_get_ptr(data, "item", &sceneItem);
+
+	const char* sceneName =
+		obs_source_get_name(obs_scene_get_source(scene));
+	const char* sceneItemName =
+		obs_source_get_name(obs_sceneitem_get_source(sceneItem));
+
+	OBSDataAutoRelease fields = obs_data_create();
+	obs_data_set_string(fields, "scene-name", sceneName);
+	obs_data_set_string(fields, "item-name", sceneItemName);
+
+	OBSDataAutoRelease posData = obs_data_create();
+	vec2 pos;
+	obs_sceneitem_get_pos(sceneItem, &pos);
+	obs_data_set_double(posData, "x", pos.x);
+	obs_data_set_double(posData, "y", pos.y);
+	obs_data_set_int(posData, "alignment", obs_sceneitem_get_alignment(sceneItem));
+	obs_data_set_obj(fields, "position", posData);
+
+	obs_data_set_double(fields, "rotation", obs_sceneitem_get_rot(sceneItem));
+
+	OBSDataAutoRelease scaleData = obs_data_create();
+	vec2 scale;
+	obs_sceneitem_get_scale(sceneItem, &scale);
+	obs_data_set_double(scaleData, "x", scale.x);
+	obs_data_set_double(scaleData, "y", scale.y);
+	obs_data_set_obj(fields, "scale", scaleData);
+
+	OBSDataAutoRelease cropData = obs_data_create();
+	obs_sceneitem_crop crop;
+	obs_sceneitem_get_crop(sceneItem, &crop);
+	obs_data_set_int(cropData, "left", crop.left);
+	obs_data_set_int(cropData, "top", crop.top);
+	obs_data_set_int(cropData, "right", crop.right);
+	obs_data_set_int(cropData, "bottom", crop.bottom);
+	obs_data_set_obj(fields, "crop", cropData);
+
+	obs_data_set_bool(fields, "visible", obs_sceneitem_visible(sceneItem));
+
+	OBSDataAutoRelease boundsData = obs_data_create();
+	obs_bounds_type boundsType = obs_sceneitem_get_bounds_type(sceneItem);
+	if (boundsType == OBS_BOUNDS_NONE) {
+		obs_data_set_string(boundsData, "type", "OBS_BOUNDS_NONE");
+	}
+	else {
+		switch (boundsType) {
+		case OBS_BOUNDS_STRETCH: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_STRETCH");
+			break;
+		}
+		case OBS_BOUNDS_SCALE_INNER: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_SCALE_INNER");
+			break;
+		}
+		case OBS_BOUNDS_SCALE_OUTER: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_SCALE_OUTER");
+			break;
+		}
+		case OBS_BOUNDS_SCALE_TO_WIDTH: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_SCALE_TO_WIDTH");
+			break;
+		}
+		case OBS_BOUNDS_SCALE_TO_HEIGHT: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_SCALE_TO_HEIGHT");
+			break;
+		}
+		case OBS_BOUNDS_MAX_ONLY: {
+			obs_data_set_string(boundsData, "type", "OBS_BOUNDS_MAX_ONLY");
+			break;
+		}
+		}
+		obs_data_set_int(boundsData, "alignment", obs_sceneitem_get_bounds_alignment(sceneItem));
+		vec2 bounds;
+		obs_sceneitem_get_bounds(sceneItem, &bounds);
+		obs_data_set_double(boundsData, "x", bounds.x);
+		obs_data_set_double(boundsData, "y", bounds.y);
+	}
+	obs_data_set_obj(fields, "bounds", boundsData);
```

Comment:
To avoid redundant code, it would be best to put this chunk of code into its own `Utils` function so that both this method and `GetSceneItemProperties` use it.

Reply:
I think this should do it. My C is very rusty. And I don't have this machine setup right for proper OBS testing just this second. (Since this looked like a straight forward copy/paste fudge job/patch).